hudi-common/src/main/java/org/apache/hudi/metadata/MetadataPartitionType.java

```diff
@@ -18,27 +18,77 @@
 
 package org.apache.hudi.metadata;
 
+import org.apache.hudi.common.config.TypedProperties;
+import org.apache.hudi.common.table.HoodieTableMetaClient;
+
 import java.util.Arrays;
 import java.util.Collections;
 import java.util.List;
 import java.util.Set;
 import java.util.stream.Collectors;
 
+import static org.apache.hudi.common.config.HoodieMetadataConfig.ENABLE;
+import static org.apache.hudi.common.config.HoodieMetadataConfig.ENABLE_METADATA_INDEX_BLOOM_FILTER;
+import static org.apache.hudi.common.config.HoodieMetadataConfig.ENABLE_METADATA_INDEX_COLUMN_STATS;
+import static org.apache.hudi.common.config.HoodieMetadataConfig.RECORD_INDEX_ENABLE_PROP;
+
 /**
  * Partition types for metadata table.
  */
 public enum MetadataPartitionType {
-  FILES(HoodieTableMetadataUtil.PARTITION_NAME_FILES, "files-"),
-  COLUMN_STATS(HoodieTableMetadataUtil.PARTITION_NAME_COLUMN_STATS, "col-stats-"),
-  BLOOM_FILTERS(HoodieTableMetadataUtil.PARTITION_NAME_BLOOM_FILTERS, "bloom-filters-"),
-  RECORD_INDEX(HoodieTableMetadataUtil.PARTITION_NAME_RECORD_INDEX, "record-index-"),
-  FUNCTIONAL_INDEX(HoodieTableMetadataUtil.PARTITION_NAME_FUNCTIONAL_INDEX_PREFIX, "func-index-");
+  FILES(HoodieTableMetadataUtil.PARTITION_NAME_FILES, "files-") {
+    @Override
+    public boolean isMetadataPartitionEnabled(TypedProperties properties) {
+      return properties.getBoolean(ENABLE.key(), ENABLE.defaultValue());
+    }
+  },
+  COLUMN_STATS(HoodieTableMetadataUtil.PARTITION_NAME_COLUMN_STATS, "col-stats-") {
+    @Override
+    public boolean isMetadataPartitionEnabled(TypedProperties properties) {
+      return properties.getBoolean(ENABLE_METADATA_INDEX_COLUMN_STATS.key(), ENABLE_METADATA_INDEX_COLUMN_STATS.defaultValue());
+    }
+  },
+  BLOOM_FILTERS(HoodieTableMetadataUtil.PARTITION_NAME_BLOOM_FILTERS, "bloom-filters-") {
+    @Override
+    public boolean isMetadataPartitionEnabled(TypedProperties properties) {
+      return properties.getBoolean(ENABLE_METADATA_INDEX_BLOOM_FILTER.key(), ENABLE_METADATA_INDEX_BLOOM_FILTER.defaultValue());
+    }
+  },
+  RECORD_INDEX(HoodieTableMetadataUtil.PARTITION_NAME_RECORD_INDEX, "record-index-") {
+    @Override
+    public boolean isMetadataPartitionEnabled(TypedProperties properties) {
+      return properties.getBoolean(RECORD_INDEX_ENABLE_PROP.key(), RECORD_INDEX_ENABLE_PROP.defaultValue());
+    }
+  },
+  FUNCTIONAL_INDEX(HoodieTableMetadataUtil.PARTITION_NAME_FUNCTIONAL_INDEX_PREFIX, "func-index-") {
+    @Override
+    public boolean isMetadataPartitionEnabled(TypedProperties properties) {
+      return false;
```

Comment:
Is this temporary?

Reply:
Yes. Currently, it's always created using sql and not write config. Added a comment and HUDI-7662 to add a new write config for func index.